{"text": "2024-05-11 06:26:26 | INFO    | __main__:stash:131 | run - {'image': '20231121', 'version': '0.10.1.post2', 'role': 'EpicPlayer', 'headless': True}\n", "record": {"elapsed": {"repr": "0:00:01.008150", "seconds": 1.00815}, "exception": null, "extra": {"image": "20231121", "version": "0.10.1.post2", "role": "EpicPlayer", "headless": true}, "file": {"name": "claim.py", "path": "/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/claim.py"}, "function": "stash", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 131, "message": "run", "module": "claim", "name": "__main__", "process": {"id": 3971, "name": "MainProcess"}, "thread": {"id": 139690533579648, "name": "MainThread"}, "time": {"repr": "2024-05-11 06:26:26.931535+08:00", "timestamp": 1715379986.931535}}}
{"text": "2024-05-11 06:26:30 | WARNING    | epic_games.agent:get_order_history:467 | Failed to get order history, cookie may have expired - {}\n", "record": {"elapsed": {"repr": "0:00:04.296573", "seconds": 4.296573}, "exception": null, "extra": {}, "file": {"name": "agent.py", "path": "/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/epic_games/agent.py"}, "function": "get_order_history", "level": {"icon": "‚ö†Ô∏è", "name": "WARNING", "no": 30}, "line": 467, "message": "Failed to get order history, cookie may have expired", "module": "agent", "name": "epic_games.agent", "process": {"id": 3971, "name": "MainProcess"}, "thread": {"id": 139690533579648, "name": "MainThread"}, "time": {"repr": "2024-05-11 06:26:30.219958+08:00", "timestamp": 1715379990.219958}}}
{"text": "2024-05-11 06:26:30 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'Circus Electrique', 'url': 'https://store.epicgames.com/en-US/p/circus-electrique'}\n", "record": {"elapsed": {"repr": "0:00:04.734418", "seconds": 4.734418}, "exception": null, "extra": {"title": "Circus Electrique", "url": "https://store.epicgames.com/en-US/p/circus-electrique"}, "file": {"name": "claim.py", "path": "/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3971, "name": "MainProcess"}, "thread": {"id": 139690533579648, "name": "MainThread"}, "time": {"repr": "2024-05-11 06:26:30.657803+08:00", "timestamp": 1715379990.657803}}}
{"text": "2024-05-11 06:26:30 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'Firestone Free Offer', 'url': 'https://store.epicgames.com/en-US/p/firestone-online-idle-rpg-bfd04b'}\n", "record": {"elapsed": {"repr": "0:00:04.734857", "seconds": 4.734857}, "exception": null, "extra": {"title": "Firestone Free Offer", "url": "https://store.epicgames.com/en-US/p/firestone-online-idle-rpg-bfd04b"}, "file": {"name": "claim.py", "path": "/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3971, "name": "MainProcess"}, "thread": {"id": 139690533579648, "name": "MainThread"}, "time": {"repr": "2024-05-11 06:26:30.658242+08:00", "timestamp": 1715379990.658242}}}
{"text": "2024-05-11 06:26:35 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/circus-electrique'}\n", "record": {"elapsed": {"repr": "0:00:09.770834", "seconds": 9.770834}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/circus-electrique"}, "file": {"name": "agent.py", "path": "/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3971, "name": "MainProcess"}, "thread": {"id": 139690533579648, "name": "MainThread"}, "time": {"repr": "2024-05-11 06:26:35.694219+08:00", "timestamp": 1715379995.694219}}}
{"text": "2024-05-11 06:27:10 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/firestone-online-idle-rpg-bfd04b'}\n", "record": {"elapsed": {"repr": "0:00:44.796922", "seconds": 44.796922}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/firestone-online-idle-rpg-bfd04b"}, "file": {"name": "agent.py", "path": "/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3971, "name": "MainProcess"}, "thread": {"id": 139690533579648, "name": "MainThread"}, "time": {"repr": "2024-05-11 06:27:10.720307+08:00", "timestamp": 1715380030.720307}}}
{"text": "2024-05-11 06:27:20 | ERROR    | __main__:run:158 | An error has been caught in function 'run', process 'MainProcess' (3971), thread 'MainThread' (139690533579648): - {}\nTraceback (most recent call last):\n  File \"/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/claim.py\", line 162, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n> File \"/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/claim.py\", line 158, in run\n    await agent.stash()\n  File \"/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/claim.py\", line 151, in stash\n    await self.claim_epic_games(context)\n  File \"/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/claim.py\", line 122, in claim_epic_games\n    await epic.claim_weekly_games(page, single_promotions)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 142, in async_wrapped\n    return await fn(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 58, in __call__\n    do = await self.iter(retry_state=retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 110, in iter\n    result = await action(retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 78, in inner\n    return fn(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 390, in <lambda>\n    self._add_action_func(lambda rs: rs.outcome.result())\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/concurrent/futures/_base.py\", line 451, in result\n    return self.__get_result()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n    raise self._exception\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 61, in __call__\n    result = await fn(*args, **kwargs)\n  File \"/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/epic_games/agent.py\", line 310, in claim_weekly_games\n    await expect(cta_btn).to_have_text(\"View In Cart\")\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py\", line 18871, in to_have_text\n    await self._impl_obj.to_have_text(\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 471, in to_have_text\n    await self._expect_impl(\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 70, in _expect_impl\n    raise AssertionError(\nAssertionError: Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\n  - waiting for locator(\"//aside//button[@data-testid='add-to-cart-cta-button']\")\n\n", "record": {"elapsed": {"repr": "0:00:54.915293", "seconds": 54.915293}, "exception": {"type": "AssertionError", "value": "Locator expected to have text 'View In Cart'\nActual value: None \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\n  - waiting for locator(\"//aside//button[@data-testid='add-to-cart-cta-button']\")\n", "traceback": true}, "extra": {}, "file": {"name": "claim.py", "path": "/home/runner/work/epic-awesome-gamer/epic-awesome-gamer/epic/src/claim.py"}, "function": "run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 158, "message": "An error has been caught in function 'run', process 'MainProcess' (3971), thread 'MainThread' (139690533579648):", "module": "claim", "name": "__main__", "process": {"id": 3971, "name": "MainProcess"}, "thread": {"id": 139690533579648, "name": "MainThread"}, "time": {"repr": "2024-05-11 06:27:20.838678+08:00", "timestamp": 1715380040.838678}}}
